### ISSUE TITLE
Bug: secrets stopped working with surfshark after pr-756?

### ISSUE THREAD

**Issue by malvim** [qdm12/gluetun#790]
### Is this urgent?

No

### Host OS

Raspbian Bullseye

### CPU arch

aarch64

### VPN service provider

Surfshark

### What are you using to run the container

docker-compose

### What is the version of Gluetun

Running version latest built on 2022-01-07T22:46:55.519Z (commit 6f5a78c)

### What's the problem 🤔

Gluetun crashes with error:

```
gluetun    | 2022/01/14 00:09:02 INFO openvpn: neither stdin nor stderr are a tty device and you have neither a controlling tty nor systemd - can't ask for 'Enter Auth Password:'.  If you used --daemon, you need to use --askpass to make passphrase-protected keys work, and you can not use --auth-nocache.
gluetun    | 2022/01/14 00:09:02 INFO openvpn: Exiting due to fatal error
```

This happens with current `latest` and `pr-756` tags.

If I switch to `v3.26.0` it starts fine, and if I use env vars instead of secrets files, it also works.

### Share your logs

```log
gluetun    | ========================================
gluetun    | ========================================
gluetun    | =============== gluetun ================
gluetun    | ========================================
gluetun    | =========== Made with ❤️ by ============
gluetun    | ======= https://github.com/qdm12 =======
gluetun    | ========================================
gluetun    | ========================================  
gluetun    |                                                                        
gluetun    | Running version latest built on 2022-01-07T22:46:55.519Z (commit 6f5a78c)
gluetun    |                                                                        
gluetun    | 🔧 Need help? https://github.com/qdm12/gluetun/discussions/new
gluetun    | 🐛 Bug? https://github.com/qdm12/gluetun/issues/new
gluetun    | ✨ New feature? https://github.com/qdm12/gluetun/issues/new                                                                                                
gluetun    | ☕ Discussion? https://github.com/qdm12/gluetun/discussions/new  
gluetun    | 💻 Email? quentin.mcgaw@gmail.com                                      
gluetun    | 💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12                                                                                 
gluetun    | 2022/01/14 00:13:37 INFO storage: merging by most recent 11100 hardcoded servers and 11100 servers read from /gluetun/servers.json
gluetun    | 2022/01/14 00:13:38 INFO Alpine version: 3.15.0                                                                                                            
gluetun    | 2022/01/14 00:13:38 INFO OpenVPN 2.4 version: 2.4.11                                                                                                       
gluetun    | 2022/01/14 00:13:38 INFO OpenVPN 2.5 version: 2.5.4                                                                                                        
gluetun    | 2022/01/14 00:13:38 INFO Unbound version: 1.13.2
gluetun    | 2022/01/14 00:13:38 INFO IPtables version: v1.8.7      
gluetun    | 2022/01/14 00:13:38 INFO Settings summary:                                                                                                                 
gluetun    | ├── VPN settings:                                                      
gluetun    | |   ├── VPN provider settings:                                         
gluetun    | |   |   ├── Name: surfshark                                                                                                                                
gluetun    | |   |   └── Server selection settings:                                                                                                                     
gluetun    | |   |       ├── VPN type: openvpn                                                                                                                          
gluetun    | |   |       └── OpenVPN server selection settings:                                                                                                         
gluetun    | |   |           └── Protocol: UDP                                                                                                                          gluetun    | |   └── OpenVPN settings:                                                   
gluetun    | |       ├── OpenVPN version: 2.5                                       
gluetun    | |       ├── User: [set]                                                
gluetun    | |       ├── Password: [set]                                            
gluetun    | |       ├── Tunnel IPv6: no                                                                                                                                
gluetun    | |       ├── Network interface: tun0             
gluetun    | |       ├── Run OpenVPN as: root
gluetun    | |       └── Verbosity level: 1
gluetun    | ├── DNS settings:
gluetun    | |   ├── DNS server address to use: 127.0.0.1
gluetun    | |   ├── Keep existing nameserver(s): no
gluetun    | |   └── DNS over TLS settings:
gluetun    | |       ├── Enabled: yes
gluetun    | |       ├── Update period: every 24h0m0s
gluetun    | |       ├── Unbound settings:
gluetun    | |       |   ├── Authoritative servers:
gluetun    | |       |   |   └── cloudflare
gluetun    | |       |   ├── Caching: yes
gluetun    | |       |   ├── IPv6: no
gluetun    | |       |   ├── Verbosity level: 1
gluetun    | |       |   ├── Verbosity details level: 0
gluetun    | |       |   ├── Validation log level: 0
gluetun    | |       |   ├── System user: root
gluetun    | |       |   └── Allowed networks:
gluetun    | |       |       ├── 0.0.0.0/0
gluetun    | |       |       └── ::/0
gluetun    | |       └── DNS filtering settings:
gluetun    | |           ├── Block malicious: yes
gluetun    | |           ├── Block ads: no
gluetun    | |           ├── Block surveillance: no
gluetun    | |           └── Blocked IP networks:
gluetun    | |               ├── 127.0.0.1/8
gluetun    | |               ├── 10.0.0.0/8
gluetun    | |               ├── 172.16.0.0/12
gluetun    | |               ├── 192.168.0.0/16
gluetun    | |               ├── 169.254.0.0/16
gluetun    | |               ├── ::1/128
gluetun    | |               ├── fc00::/7
gluetun    | |               ├── fe80::/10
gluetun    | |               ├── ::ffff:7f00:1/104
gluetun    | |               ├── ::ffff:a00:0/104
gluetun    | |               ├── ::ffff:a9fe:0/112
gluetun    | |               ├── ::ffff:ac10:0/108
gluetun    | |               └── ::ffff:c0a8:0/112
gluetun    | ├── Firewall settings:                                                                                                                             [4/2506]
gluetun    | |   └── Enabled: yes
gluetun    | ├── Log settings:
gluetun    | |   └── Log level: INFO
gluetun    | ├── Health settings:
gluetun    | |   ├── Server listening address: 127.0.0.1:9999
gluetun    | |   ├── Address to ping: github.com
gluetun    | |   └── VPN wait durations:
gluetun    | |       ├── Initial duration: 5s
gluetun    | |       └── Additional duration: 5s
gluetun    | ├── Shadowsocks server settings:
gluetun    | |   └── Enabled: no
gluetun    | ├── HTTP proxy settings:
gluetun    | |   └── Enabled: no
gluetun    | ├── Control server settings:
gluetun    | |   ├── Listening port: 8000
gluetun    | |   └── Logging: yes
gluetun    | ├── OS Alpine settings:
gluetun    | |   ├── Process UID: 1000
gluetun    | |   ├── Process GID: 1000
gluetun    | |   └── Timezone: America/Sao_Paulo
gluetun    | ├── Public IP settings:
gluetun    | |   ├── Fetching: every 12h0m0s
gluetun    | |   └── IP file path: /tmp/gluetun/ip
gluetun    | └── Version settings:
gluetun    |     └── Enabled: yes
gluetun    | 2022/01/14 00:13:38 INFO routing: default route found: interface eth0, gateway 172.17.0.1
gluetun    | 2022/01/14 00:13:38 INFO routing: local ethernet link found: eth0
gluetun    | 2022/01/14 00:13:38 INFO routing: local ipnet found: 172.17.0.0/16
gluetun    | 2022/01/14 00:13:38 INFO routing: default route found: interface eth0, gateway 172.17.0.1
gluetun    | 2022/01/14 00:13:38 INFO routing: adding route for 0.0.0.0/0
gluetun    | 2022/01/14 00:13:38 INFO firewall: firewall disabled, only updating allowed subnets internal list
gluetun    | 2022/01/14 00:13:38 INFO routing: default route found: interface eth0, gateway 172.17.0.1
gluetun    | 2022/01/14 00:13:38 INFO TUN device is not available: open /dev/net/tun: no such file or directory; creating it...
gluetun    | 2022/01/14 00:13:38 INFO firewall: enabling...
gluetun    | 2022/01/14 00:13:38 INFO firewall: enabled successfully
gluetun    | 2022/01/14 00:13:38 INFO firewall: enabled successfully
gluetun    | 2022/01/14 00:13:38 INFO dns over tls: using plaintext DNS at address 1.1.1.1
gluetun    | 2022/01/14 00:13:38 INFO http server: listening on :8000
gluetun    | 2022/01/14 00:13:38 INFO healthcheck: listening on 127.0.0.1:9999
gluetun    | 2022/01/14 00:13:38 INFO openvpn configurator: username and password changed in /etc/openvpn/auth.conf
gluetun    | 2022/01/14 00:13:38 INFO firewall: setting VPN connection through firewall...
gluetun    | 2022/01/14 00:13:38 INFO openvpn: OpenVPN 2.5.4 aarch64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Nov 15 2021
gluetun    | 2022/01/14 00:13:38 INFO openvpn: library versions: OpenSSL 1.1.1l  24 Aug 2021, LZO 2.10
gluetun    | 2022/01/14 00:13:38 INFO openvpn: neither stdin nor stderr are a tty device and you have neither a controlling tty nor systemd - can't ask for 'Enter Auth 
Password:'.  If you used --daemon, you need to use --askpass to make passphrase-protected keys work, and you can not use --auth-nocache.
gluetun    | 2022/01/14 00:13:38 INFO openvpn: Exiting due to fatal error
gluetun    | 2022/01/14 00:13:38 ERROR vpn: exit status 1
gluetun    | 2022/01/14 00:13:38 INFO vpn: retrying in 15s
gluetun    | 2022/01/14 00:13:43 INFO healthcheck: program has been unhealthy for 5s: restarting VPN
```

### Share your configuration

```yml
version: "3.7"
secrets:
  openvpn_user:
    file: <SURFSHARK_USER_FILE>
  openvpn_password:
    file: <SURFSHARK_PASSWORD_FILE>
services:
  gluetun:
    image: qmcgaw/gluetun:latest
    container_name: gluetun
    cap_add:
      - NET_ADMIN
    network_mode: bridge
    ports:
      - 8888:8888/tcp # HTTP proxy
      - 8388:8388/tcp # Shadowsocks
      - 8388:8388/udp # Shadowsocks
    volumes:
      - ./gluetun:/gluetun
    secrets:
      - openvpn_user
      - openvpn_password
    environment:
      - VPNSP=surfshark
    restart: always
```

**Comment by qdm12** [comment]
That's strange because it logs out `Password: [set]` so it's definitely not the empty string. The file probably got read as well without error I would say.

Are you sure the password secret file is not empty?

**Comment by qdm12** [comment]
Just tried it and yes it doesn't work, I'm working on it.

**Comment by qdm12** [comment]
Fixed in a1a1128d6de28aa720492d5502f754e315d4ad54

It was because it was now reading the entire file content, even the trailing new line.
You probably had a trailing new line in your secret file 😉 
Anyway I reverted it to trim `\r\n` and `\n` suffixes from files since I doubt there would be an actual use case to have a password ending with one of those.
